### PR TITLE
[sanity check] Fix and improve recovery

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
The sanity check could fail during recovery on single-asic platform when there are interfaces down. 

#### How did you do it?
This change fixed that issue by firstly checking if the DUT is multi-asic or not. This change also made some other improvements:
* If there are bgp session down, try to "no shutdown" bgp session on neighbors and then do a config reload.
* If there are interface down, previously the code will only run "no shutdown" of the corresponding port on fanout switch. This change added an extra "shutdown" step before "no shutdown".
* Exclude "FLEX_COUNTER_TABLE" from post test config comparison.
* Improve logs and function names for the fixture checking core dumps and comparing config after testing.

#### How did you verify/test it?
Tested in KVM and physical testbeds.

### CODE REVIEW

File: tests/common/plugins/sanity_check/recover.py [hunk 1]
```diff
@@ -149,5 +103,58 @@ def neighbor_vm_restore(duthost, nbrhosts, tbinfo):
     mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
     vm_neighbors = mg_facts['minigraph_neighbors']
     if vm_neighbors:
-        results = parallel_run(neighbor_vm_recover_bgpd, (), {}, nbrhosts.values(), timeout=300)
+        results = parallel_run(__neighbor_vm_recover_bgpd, (), {}, nbrhosts.values(), timeout=300)
         logger.debug('Results of restoring neighbor VMs: {}'.format(json.dumps(dict(results))))
+    return 'config_reload'  # May still need to do a config reload
+
+
+def __recover_with_command(dut, cmd, wait_time):
```
[the remark this answers is not in the document]
Comment (wangxin) [reply tests/common/plugins/sanity_check/recover.py]: Sure, updated.

**Review by lolyu — APPROVED** [review]

**Comment by mithun2498** [comment]
Hello @wangxin ,

I tried executing T1 topology cases in physical setup and I could see many cases are failing due to "Recovery of sanity check failed". Kindly suggest how to resolve the above issue.

**Comment by wangxin** [comment]
@mithun2498 Please check the detailed log to find out why sanity check failed. Sanity check recovery is not able to recover all issue. For example, if ptf or neighbor ceos containers are down, sanity check recovery is not able to recover. In this case, need to redeploy the testbed topology.

**Pull request merged**